syntaxes/pyright.sublime-syntax

```diff
@@ -2,150 +2,167 @@
+    - match: \)
+      scope: meta.class.parameters.python punctuation.section.parameters.end.python
+      pop: 1
+    - match: ','
+      scope: punctuation.separator.parameters.python
+      push: allow-unpack-operators
+    - match: '{{assignment_operator}}'
+      scope: keyword.operator.assignment.python
+      set: class-definition-parameter-default-value
+    - match: '{{colon}}'
+      scope: punctuation.separator.annotation.parameter.python
+      set: class-definition-parameter-annotation
+    - include: parameter-names
+    - include: line-continuations
+    - include: illegal-assignment-expressions
+
+  class-definition-parameter-annotation:
+    - meta_scope: meta.class.parameters.annotation.python
+    - match: (?=[,)=])
+      set: class-definition-base-list-body
+    - match: None\b
+      scope: constant.language.null.python
+    - include: expression-in-a-group
+
+  class-definition-parameter-default-value:
+    - meta_scope: meta.class.parameters.default-value.python
+    - match: (?=[,)])
+      set: class-definition-base-list-body
+    - include: expression-in-a-group
+
+###[ PYTHON FUNCTION DEFINITION OVERRIDES ]###################################
+
+  function-definitions:
+    # make sure to highlight `def` after pyright's `(function)` tag.
+    - match: (?:(async)\s+)?(def)\b
+      scope: meta.function.python
       captures:
         1: keyword.declaration.async.python
         2: keyword.declaration.function.python
-      push:
-        - meta_scope: meta.function.python
-        - include: line-continuation-or-pop
-        - match: ':'
-          scope: punctuation.section.function.begin.python
-          pop: true
-        - match: '(?={{identifier}})'
-          push:
-            - meta_content_scope: entity.name.function.python
-            - include: entity-name-function
-            - match: ''
-              pop: true
-        - match: '(?=\()'
-          set:
-            - match: \(
-              scope: meta.function.parameters.python punctuation.section.parameters.begin.python
-              set: [function-parameters, allow-unpack-operators]
-
-  function-terminator:
+      push: function-definition-name
+
+  function-definition-name:
+    - meta_prepend: true
+    # colon without function parameter list denotes extended function signature
+    - match: '{{colon}}'
+      scope: meta.function.python punctuation.section.function.begin.python
+      set: lsp-function-signature
+
+  lsp-function-signature:
+    - meta_content_scope: meta.function.annotation.signature.python
+    - include: function-definition-end
+    - include: lsp-type-annotations
+    - include: expression-in-a-statement
+
+  function-definition-end:
+    - meta_prepend: true
     # match trailing ending brackets in LSP popup
-    - match: \)
-      scope: punctuation.section.parameters.end.python
-      pop: true
-    - match: \]
-      scope: punctuation.section.sequence.end.python
-      pop: true
+    - match: (?=[])])
```

Comment:
It seems working but I personally prefer escaping `]`.

```suggestion
    - match: (?=[\])])
```

Reply:
And I was so proud to have saved one char 🤣 .